src/main/java/com/rabbitmq/client/impl/TruncatedInputStream.java

```diff
@@ -42,7 +42,7 @@ public int available() throws IOException {
     }
 
     @Override
-    public void mark(int readlimit) {
+    public synchronized void mark(int readlimit) {
```

Comment:
the super mark/reset methods are synchronized, this is addressing a findbug warning